### PR TITLE
[PLAYER-4984] changes after eslint

### PR DESCRIPTION
**Pull request by MariskaS**

### CODE REVIEW

File: js/ad_manager_vast.js [hunk 1]
```diff
@@ -483,6 +500,7 @@ OO.Ads.manager(() => {
      * @returns {boolean} true if the skip ad functionality is supported in the specified Vast version,
      *                    false otherwise.
      */
+    // eslint-disable-next-line max-len
     const supportsSkipAd = version => contains(SUPPORTED_FEATURES[getMajorVersion(version)], FEATURES.SKIP_AD);
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: just split it to several lines
```
const supportsSkipAd = version => contains(
  SUPPORTED_FEATURES[getMajorVersion(version)], 
  FEATURES.SKIP_AD,
);
```
here and below

File: js/pulse.js [hunk 2]
```diff
@@ -147,14 +147,14 @@
         const pos = parseInt(position);
         const insertionPointFilter = [];
 
-        if (pos & PREROLL) {
+        if (pos && PREROLL) {
```
Comment (ooyala-amohnatkin) [inline js/pulse.js]: `&` and `&&` are very different, please revert and disable the rule for this block

Comment (MariskaS) [reply js/pulse.js]: fixed within another commit

File: js/ad_manager_template.js [hunk 3]
```diff
@@ -241,27 +218,11 @@ OO.Ads.manager(() => {
      * the initialize function.
      * @public
      * @method AdManager#onVideoTagFound
-     * @param {string} event The event that triggered this callback.
-     * @param {string} videoId The id of the video element that processed a tag.
-     * @param {string} tagType The type of tag that was detected.
-     * @param {object} metadata Any metadata attached to the found tag.
      */
-    this.onVideoTagFound = (event, videoId, tagType, metadata) => {
+    this.onVideoTagFound = () => {
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_template.js]: please revert it and pass named arguments in `OO.log`

File: test/unit-tests/vast_test.js [hunk 4]
```diff
@@ -2169,6 +2170,7 @@ describe('ad_manager_vast', function () {
         {
           tag_url: 'http://blahblah',
           position_type: 'p',
+          // eslint-disable-next-line require-jsdoc
```
Comment (ooyala-amohnatkin) [inline test/unit-tests/vast_test.js]: this rule is disabled for the whole file, you can remove this line

File: js/ad_manager_template.js [hunk 5]
```diff
@@ -35,20 +33,15 @@ OO.Ads.manager(() => {
      * @param {object} adManagerController A reference to the Ad Manager Controller
      * @param {string} playerId The unique player identifier of the player initializing the class
      */
-    this.initialize = (adManagerController, playerId) => {
+    this.initialize = (adManagerController) => {
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_template.js]: Not sure about this file. This is a boilerplate for the ad manager and it should be like an example of how to implement it. Maybe it's better to revert and disable some rules for this file. @ooyala/player-web ?

File: js/ad_manager_vast.js [hunk 6]
```diff
@@ -1747,31 +1549,6 @@ OO.Ads.manager(() => {
...
 
-    /**
-     * When the Ad Manager Controller needs to hide the overlay it will call this function. We will store the current ad
-     * for reference. Vast ad doesn't need to do much other then save the reference.
-     * @public
-     * @method Vast#hideOverlay
-     * @param {object} currentAd In order to not lose reference to the overlay object that is currently being shown, it
-     * is stored in this object
-     */
-    this.hideOverlay = (currentAd) => {
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: This is a public method and it's called from ad manager controller. You shouldn't remove public methods

File: js/ad_manager_vast.js [hunk 7]
```diff
@@ -1849,8 +1626,8 @@ OO.Ads.manager(() => {
      * @param {string} url URL to ping
      */
     this.pingURL = (code, url) => {
-      url = url.replace(/\[ERRORCODE\]/, code);
-      OO.pixelPing(url);
+      const urlNew = url.replace(/\[ERRORCODE\]/, code);
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: please make a variable name more verbose, like `pingUrl`

File: js/ad_manager_vast.js [hunk 8]
```diff
@@ -1928,7 +1705,7 @@ OO.Ads.manager(() => {
         return null;
       }
 
-      params = params || {};
+      const paramsObj = params || {};
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: please use default parameters instead
`const _handleLinearAd = (ad, adLoaded, params = {}) => {`

File: js/ad_manager_vast.js [hunk 9]
```diff
@@ -1981,15 +1757,15 @@ OO.Ads.manager(() => {
         this.trackError(this.ERROR_CODES.GENERAL_NONLINEAR_ADS, ad.id);
         return null;
       }
-      params = params || {};
+      const paramsObj = params || {};
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: please use default parameters

File: js/ad_manager_vast.js [hunk 10]
```diff
@@ -2509,8 +2161,9 @@ OO.Ads.manager(() => {
             }
           } else {
             processedFallbackAd = fallbackAd;
-          }
+          } // eslint-disable-next-line no-undef
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: why disable this rule here and below?

File: js/ad_manager_vast.js [hunk 11]
```diff
@@ -1549,6 +1549,31 @@ OO.Ads.manager(() => {
...
+    /**
+     * When the Ad Manager Controller needs to hide the overlay it will call this function. We will store the current ad
+     * for reference. Vast ad doesn't need to do much other then save the reference.
+     * @public
+     * @method Vast#hideOverlay
+     * @param {object} currentAd In order to not lose reference to the overlay object that is currently being shown, it
+     * is stored in this object
+     */
+    this.hideOverlay = (currentAd) => {
+      this.lastOverlayAd = currentAd;
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: `this.lastOverlayAd` was removed too. Revert it too and check that you didn't miss anything else

File: js/ad_manager_vast.js [hunk 12]
```diff
@@ -894,7 +894,7 @@ OO.Ads.manager(() => {
      */
     const loadAd = (amcAd) => {
       let loadedAds = false;
-      const { ad } = amcAd;
+      const {ad} = amcAd;
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: Why did you remove spaces? Eslint will fail here

File: js/ad_manager_template.js [hunk 13]
```diff
@@ -25,6 +25,7 @@ OO.Ads.manager(() => {
 
     let amc = null;
     const remoteModuleJs = 'http://my.company/myAdModule.js';
+    // eslint-disable-next-line no-unused-vars
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_template.js]: it's used on the line 51

File: js/ad_manager_template.js [hunk 14]
```diff
@@ -241,13 +225,9 @@ OO.Ads.manager(() => {
...
      * @public
      * @method AdManager#onVideoTagFound
-     * @param {string} event The event that triggered this callback.
-     * @param {string} videoId The id of the video element that processed a tag.
-     * @param {string} tagType The type of tag that was detected.
-     * @param {object} metadata Any metadata attached to the found tag.
      */
-    this.onVideoTagFound = (event, videoId, tagType, metadata) => {
-      OO.log('TAG FOUND w/ args: ', arguments);
+    this.onVideoTagFound = (...args) => {
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_template.js]: please revert parameters and jsdoc for them

File: js/ad_manager_template.js [hunk 15]
```diff
@@ -35,11 +36,12 @@ OO.Ads.manager(() => {
      * @param {object} adManagerController A reference to the Ad Manager Controller
      * @param {string} playerId The unique player identifier of the player initializing the class
      */
+    // eslint-disable-next-line no-unused-vars
     this.initialize = (adManagerController, playerId) => {
       amc = adManagerController;
 
       // Add any player event listeners now
-      amc.addPlayerListener(amc.EVENTS.CONTENT_CHANGED, _onContentChanged);
+      amc.addPlayerListener(amc.EVENTS.CONTENT_CHANGED);
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_template.js]: the second parameter should be a function, please revert `_onContentChanged`

File: js/ad_manager_vast.js [hunk 16]
```diff
@@ -54,8 +54,8 @@ OO.Ads.manager(() => {
    * maximum is reached
    * @property {boolean} loaded Set to true once the ad has been loaded successfully
    * @property {string} embedCode Keeps track of the embed code of the movie that is currently playing
-   * @property {object} lastOverlayAd Contains the ad information for the overlay that was displayed before it was removed.
    * This is used so we know what to add back to the screen after the video ad is done and the main video hasn't ended.
+   * @property {object} lastOverlayAd Contains the ad information for the overlay that was displayed before it was removed.
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: previous line should go after this one

File: js/ad_manager_vast.js [hunk 17]
```diff
@@ -74,7 +74,7 @@ OO.Ads.manager(() => {
     this.currentDepth = 0;
     this.loaded = false;
     this.embedCode = 'unknown';
-    this.lastOverlayAd;
+    this.lastOverlayAd = {};
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: it was undefined, please revert

File: js/ad_manager_vast.js [hunk 18]
```diff
@@ -3630,62 +3400,331 @@ OO.Ads.manager(() => {
...
-    // TODO: consider error message override
-    var _safeFunctionCall = (vpaidAd, funcName, params) => {
-      try {
-        if (isFunction(vpaidAd[funcName])) {
-          return vpaidAd[funcName].apply(vpaidAd, params);
+    /**
+     * Callback when the media file is loaded. Once is loaded we can initialize the ad
+     * This is only required for VPAID ads
+     * @public
+     * @returns {null} Returns null
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: It shouldn't return anything, just `void`. Please correct `return null;` to just `return;`

File: test/unit-tests/ima_test.js [hunk 19]
```diff
@@ -1,32 +1,36 @@
...
 describe('ad_manager_ima', function () {
-  let amc; let
+  let amc;
+  let
     ima;
   let imaVideoPluginFactory;
   let videoWrapper;
   let imaIframe;
-  const name = 'google-ima-ads-manager';
+  const nameAdsManager = 'google-ima-ads-manager';
```
Comment (ooyala-amohnatkin) [inline test/unit-tests/ima_test.js]: adManagerName

File: js/ad_manager_template.js (hunk 13 above)
Comment (MariskaS) [reply js/ad_manager_template.js]: agreed with @alexeyMohnatkin that this is the best option

File: js/liverail.js [hunk 20]
```diff
@@ -29,7 +30,12 @@ OO.Ads.manager(() => {
     AD_LOG: 'AdLog',
   };
 
-  const Liverail = function () {
+  /**
+   * @class Liverail
+   * @classDesc The Liverail class.
+   * @constructor
+   */
+  const Liverail = function (...args) {
```
Comment (ooyala-amohnatkin) [inline js/liverail.js]: You try to use these arguments instead of arguments in each particular function. Please, use destructure in functions, not in the constructor

File: js/liverail.js [hunk 21]
```diff
@@ -49,107 +55,200 @@ OO.Ads.manager(() => {
...
     let countdownIntervalId = null;
 
     // /// Helpers /////
+    /**
+     * Log.
+     */
     const log = () => {
       if (isFunction(OO.log)) {
-        OO.log.apply(null, ['liverail-ads-manager:'].concat(Array.prototype.slice.apply(arguments)));
+        OO.log.apply(null, ['liverail-ads-manager:'].concat(args));
```
Comment (ooyala-amohnatkin) [inline js/liverail.js]: `args` come from the closure, please take them from `log` function

File: js/liverail.js [hunk 22]
```diff
@@ -49,107 +55,200 @@ OO.Ads.manager(() => {
...
       } else {
-        console.log(['liverail-ads-manager:'].concat(Array.prototype.slice.apply(arguments)));
+        console.log(['liverail-ads-manager:'].concat(args));
```
Comment (ooyala-amohnatkin) [inline js/liverail.js]: `args` come from the closure, please take them from `log` function

File: js/liverail.js [hunk 23]
```diff
@@ -49,107 +55,200 @@ OO.Ads.manager(() => {
...
-      _tryInit();
+    /**
+     * On Ad Event.
+     * @param {string} eventName The name of event.
+     * @param {*} logData ...
+     * @private
+     */
+    const _onAdEvent = (eventName, logData) => {
+      if (eventName !== VPAID_EVENTS.AD_LOG) {
+        log(eventName, 'fired with args', args.slice(1));
```
Comment (ooyala-amohnatkin) [inline js/liverail.js]: `args` come from the closure, please take them from `_onAdEvent ` function

File: js/ad_manager_vast.js [hunk 24]
```diff
@@ -379,51 +395,76 @@ OO.Ads.manager(() => {
...
+
+      // if arguments are comma separated we want to leverage console.log's ability to
+      // pretty print objects rather than printing an object's toStr representation.
+      // TODO: print this log in amc.raiseAdError
+      if (arguments.length > 1) {
+        OO.log.apply(OO.log, args);
+
+        // converts the arguments keyword to an Array.
+        // arguments looks like an Array, but isn't.
+        const convertArgs = [].slice.call(args);
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: if you use destructure they are already `Array`

File: js/ad_manager_vast.js [hunk 25]
```diff
@@ -3559,6 +3170,180 @@ OO.Ads.manager(() => {
...
+
+    /**
+     * Listen and executes events sent by the ad unit
+     * This is only required for VPAID ads
+     * @private
+     * @method Vast#_onVpaidAdEvent
+     * @param {array} args The array of arguments (eventName - Name of the event to process).
+     */
+    const _onVpaidAdEvent = function (...args) {
+      const [eventName] = args;
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: define `eventName`, `url` and `playerHandles` in the function parameters

File: js/ooyala_ssai.js [hunk 26]
```diff
@@ -548,33 +260,11 @@ OO.Ads.manager(() => {
      * @public
      * @method OoyalaSsai#onReplay
      */
-    this.onReplay = function()
-    {
+    this.onReplay = function () {
```
Comment (ooyala-amohnatkin) [inline js/ooyala_ssai.js]: why not arrow function?

File: js/pulse.js [hunk 27]
```diff
@@ -6,9 +6,11 @@
   const pulseAdManagers = {};
 
 
-  OO.Ads.manager(function () {
+  OO.Ads.manager((...args) => {
```
Comment (ooyala-amohnatkin) [inline js/pulse.js]: The same mistake with arguments as in the `liverail.js`. Please fix all appearances

File: js/ssai_heartbeat_ooyala.js [hunk 28]
```diff
@@ -76,6 +73,20 @@ OO.plugin('heartbeat', (OO) => {
...
+     */
+    function startHeartBeat() {
+      stopHeartBeat();
+      heartbeatTimer = setInterval(reportHeartBeat, config.Interval);
+    }
+
+    /**
+     * Parse Guid.
+     * @param {string} url The streamUrl.
+     * @returns {''|string} Returns '' or first match result.
```
Comment (ooyala-amohnatkin) [inline js/ssai_heartbeat_ooyala.js]: `''` is a string too
The description could be more verbose
`@returns {string} Guid extracted from URL or empty string`

File: js/ssai_heartbeat_ooyala.js [hunk 29]
```diff
@@ -95,29 +114,62 @@ OO.plugin('heartbeat', (OO) => {
...
     function _onPlayed() {
       reportingPaused = false;
       reportHeartBeat();
       stopHeartBeat();
     }
 
+    /**
+     * Build Config.
+     * @param {object} configuration The configuration object
+     * @returns {object} Returns _config object
```
Comment (ooyala-amohnatkin) [inline js/ssai_heartbeat_ooyala.js]: No need to duplicate word `Returns`

File: js/vast_pings.js [hunk 30]
```diff
@@ -60,39 +87,71 @@
...
 
-    onPlayheadTimeChanged(event, time, duration) {
-      if (this.currentVastAd == null || duration == 0) { return; }
+    /**
+     * Callback for when we receive the PLAYHEAD_TIME_CHANGED event from the AMC.
+     * @method VastPings#onPlayheadTimeChanged
+     * @param {string} eventName The name of the event for which this callback is called
+     * @param {number} time The total amount main video playback time (seconds)
+     * @param {number} duration Duration of the live video (seconds).
+     * @public
```
Comment (ooyala-amohnatkin) [inline js/vast_pings.js]: it's private

File: .eslintrc.js [hunk 31]
```diff
@@ -4,56 +4,81 @@ module.exports = {
...
-    "no-continue": "warn",
-    "brace-style": "warn",
-    "no-unused-expressions": "warn",
-    "no-lonely-if": "warn",
-    "prefer-spread": "warn",
-    "operator-linebreak": "warn",
-    "no-fallthrough": "warn",
-    "no-mixed-operators": "warn",
-    "no-loop-func": "warn",
+    "no-underscore-dangle": "warn",
```
Comment (ooyala-amohnatkin) [inline .eslintrc.js]: There are still warnings in eslint config.

File: js/ad_manager_vast.js [hunk 32]
```diff
@@ -451,7 +447,7 @@ OO.Ads.manager(() => {
      * @param {object} vpaidAd The vpaidAd object.
      * @param {string} funcName The function name.
      * @param {array} params The array of params.
-     * @returns {null|function} Returns function with vpaidAd and params arguments.
+     * @returns {null|function} function with vpaidAd and params arguments.
```
Comment (ooyala-amohnatkin) [inline js/ad_manager_vast.js]: It doesn't return function, just executes and return the result. Please fix to `*`

File: js/freewheel.js [hunk 33]
```diff
@@ -315,7 +315,7 @@ OO.Ads.manager(() => {
      * Freewheel assumes that the ad video element is an html5 video tag.  To force use of this element,
      * always list the stream type as mp4.
      * @method Freewheel#_prepareTimeline
-     * @returns {null|array} Returns [] or null
+     * @returns {null|array} [] or null
```
Comment (ooyala-amohnatkin) [inline js/freewheel.js]: it seems that we don't use the result of this function, so there's no need to return anything, just `void`

File: js/ooyala_ssai.js [hunk 34]
```diff
@@ -984,7 +984,7 @@ OO.Ads.manager(() => {
      * Callback used when the duration of an ad has passed.
      * @param {number} clearTimeoutId The clearTimeoutId for prevents a timeout from triggering.
      * @param {string} objectId The property name of adIdDictionary object.
-     * @returns {function()} Return function.
+     * @returns {function()} function.
```
Comment (ooyala-amohnatkin) [inline js/ooyala_ssai.js]: it doesn't return anything

File: .eslintrc.js [hunk 35]
```diff
@@ -4,56 +4,86 @@ module.exports = {
...
+      }
+    ],
+    "no-param-reassign": [
+      "error",
+      {
+        "props": false
+      }
+    ],
+    "no-underscore-dangle": 0,
+    // "consistent-return": [
```
Comment (KseniiaSokolovaEpam) [inline .eslintrc.js]: Do I need to remove the commented part?
